.travis.yml

```diff
@@ -37,8 +37,10 @@ install:
 
 script:
   - cd tests
-  - docker-compose config --services | grep $PLATFORM | xargs docker-compose up -d
-  - ansible-playbook $PLAYBOOK -l $PLATFORM
+  - docker-compose config --services | grep "$PLATFORM" | xargs docker-compose up -d
+  - ansible-playbook $PLAYBOOK -l "$PLATFORM:&latest" --check --diff
+  - ansible-playbook $PLAYBOOK -l "${PLATFORM:-latest}" -vv
+  - ansible-playbook $PLAYBOOK -l "$PLATFORM:&latest" --check --diff
```

Comment:
why run check mode twice?

Reply:
Running **before** and **after** the effective ansible-playbook run verifies the check mode in two completely different situations. In the current `root.yml` failure (See "Symlink ruby related binaries on the system path" task), the problem only occurs when running `--check` before the role is applied.

It is to project maintainers to decide if they want that `--check` mode should work on an "empty" or "incompletely configured" system.

Reply:
Ok I see now. Can we add a comment to explain this?

I think we definitely want `--check` mode to pass, before we apply changes, as I think this is the more common workflow, at least for me and from what I understand. And I think it seems redundant to run again after normal mode, although I'm not entirely sure either way.

Reply:
I should add: I see what you mean about running `--check` mode a second time because the system is in a different state, but I think it's up to the normal mode test run to make sure any changes are good at this point.

Reply:
My second "--check mode" run is intended to guarantee that you can execute the role in check-mode without error.

But I get you're point, and actually it made me realize that there is something _more important_ to verify: role impotency, by executing **twice** the tasks (not in check mode). I'll put some more propositions... stay tuned 😉

Reply:
@danochoa please, look at c999c43. Without this change, the second part (rubies.yml) of the role would fail.
@pkuczynski @thbar I defer to you to decide if the 'check mode' should pass on a "non-bootstrapped" system (i.e. RVM not available yet).

Note: Proposal for the "full idempotency" check comes...

Reply:
Would you mind adding a comment in the code here too to explain running check mode twice? I think that would be helpful for someone looking at the code for the first time without reading our PR comments.